GameServerCore/Domain/GameObjects/IBuff.cs

```diff
@@ -43,7 +43,7 @@ public interface IBuff : IStackable, IUpdate
         /// <summary>
         /// Unit which applied this buff to its target.
         /// </summary>
-        IObjAiBase SourceUnit { get; }
+        IAttackableUnit SourceUnit { get; }
```

Comment:
`SourceUnit` can be IAttackableUnit?

Reply:
I would not recommend this change as it sets the precedent that AttackableUnit can affect other units, which imo should only be true for position/movement.

Reply:
That had to be changed in order for the changes in "from" in AddBuff and "target" in RemoveBuff work properly.
Any suggestion on another path that could be taken if it's so problematic?

Reply:
You should be able to revert the `from` in `AddBuff` to `ObjAiBase`.

Reply:
Reverted all IObjAiBases besides in the RemoveBuff and fixed the issues in other scripts.